pkg/apis/calico/types_network.go

```diff
@@ -26,38 +26,51 @@ const (
 	VXLan Backend = "vxlan"
 )
 
-type IPv4PoolMode string
+type PoolMode string
 
 const (
-	Always      IPv4PoolMode = "Always"
-	Never       IPv4PoolMode = "Never"
-	CrossSubnet IPv4PoolMode = "CrossSubnet"
-	Off         IPv4PoolMode = "Off"
+	Always      PoolMode = "Always"
+	Never       PoolMode = "Never"
+	CrossSubnet PoolMode = "CrossSubnet"
+	Off         PoolMode = "Off"
 )
 
 type CIDR string
 
-type IPv4Pool string
+type Pool string
 
 const (
-	PoolIPIP  IPv4Pool = "ipip"
-	PoolVXLan IPv4Pool = "vxlan"
+	PoolIPIP  Pool = "ipip"
+	PoolVXLan Pool = "vxlan"
 )
 
 // IPv4 contains configuration for calico ipv4 specific settings
 type IPv4 struct {
 	// Pool configures the type of ip pool for the tunnel interface.
 	// https://docs.projectcalico.org/v3.8/reference/node/configuration#environment-variables
-	Pool *IPv4Pool
+	Pool *Pool
```

Comment:
Do we really change the API here?

Reply:
No it's basically just renamed. The only new thing for the API is the IPv6 stuff

Reply:
Ah okay I see. I figured in a IPV4 struct having the PoolMode to be named `IPv4PoolMode` is just word clutter. It also deduped some code and uses less "bloat".

However I don't mind reverting

Reply:
I would vote for keeping the short type names (without `IPv4`) as it fits the use across IPv4 and IPv6.
To keep the go API package backward-compatible, I suggest adding type aliases for `IPv4{Pool,PoolMode}`. That should be good enough :)

Reply:
OK, I'm fine with this.